libvast/vast/detail/generator.hpp

```diff
@@ -0,0 +1,252 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
```

Comment:
I think somewhere in this file we should mention that `generator<T>` works synchronously, which is crucial for the way we use it.

Maybe a link to https://github.com/lewissbaker/cppcoro#generatort would also be helpful.

Reply:
Isn't this a given because coroutines are all about concurrency? They never imply parallelism.

But yes, it probably makes sense to do add a reminder about the difference between concurrency and parallelism somewhere at the top.